pypsa/statistics.py

```diff
@@ -898,48 +980,32 @@ def energy_balance(
             Note that for {'mean', 'sum'} the time series are aggregated to MWh
```

Comment:
Maybe make the energy balance compatible with attributes of other statistic functions and make the attribute aggregate_bus setting the grouper to the corresponding one.